thirdweb-dev/js
# Improve error logging when authentication fails

## Problem solved

- When the server returns a status code for `fetchKeyMetadataFromApi` and `fetchAccountFromApi`, if it's >=400, we used to swallow the error message and arbitrarily convert it to a generic 500
- This code properly returns the auth status code/message to the caller

## Changes made


## How to test

- [x] Manual tests: step by step instructions on how to test

1. Attempt to upload to storage-server using a random (fake) Bearer token. Notice that storage-server will get a 500 back from the service-utils SDK
2. Now configure storage-server to use this version of the SDK: `0.0.0-dev-f9e7a8f-20230809203233`
3. Notice that storage-server properly receives 401 for passing an invalid bearer token
